hack/generator/pkg/astmodel/enum_type.go

```diff
@@ -34,20 +32,14 @@ func NewEnumType(baseType *PrimitiveType, options []EnumValue) *EnumType {
 
 // AsType implements Type for EnumType
 func (enum *EnumType) AsType() ast.Expr {
-	return ast.NewIdent(enum.canonicalName.name)
+	// TODO: should warn here, emitting a non-named enum
+	// when this is feature-complete this should "never" happen
+	return enum.BaseType.AsType()
```

Comment:
Won't this just be emitting `string` or `int` - is this correct?

Reply:
It should "never" happen as we should have named the enum and substituted-out the EnumType before this is called (so the `AsType` here won't be called). I added a warning to print out if it happens.